### PR TITLE
Disable connection pooling for docker client.

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
By default, `hyper` uses connection pool that can grow till `usize::MAX`. This, we believe, could cause some issues when heavy [GetModuleLogs](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-retrieve-iot-edge-logs?view=iotedge-2020-11) are frequently executed.

Also, connection pool does not have benefits for local UDS connections anyways.

### CODE REVIEW

**Review by vipeller — DISMISSED** [review]

**Comment by arsing** [comment]
For master it might be better to make a function on `http_common::Connector` that's like `fn into_client(self) -> hyper::Client`. The `Connector` knows it's a `::Unix` and can then set the pool limit. That way we get the benefit for all UDS clients in one shot.

**Comment by vadim-kovalyov** [comment]
> http_common::Connector

**Comment by vadim-kovalyov** [comment]
> For master it might be better to make a function on `http_common::Connector` that's like `fn into_client(self) -> hyper::Client`. The `Connector` knows it's a `::Unix` and can then set the pool limit. That way we get the benefit for all UDS clients in one shot.

@arsing, looks like `http_common` is in i-i-s repo. Do you want me to make changes there?

**Comment by arsing** [comment]
Yes.

File: edgelet/Cargo.lock [hunk 1]
```diff
@@ -87,7 +87,7 @@ checksum = "cdb031dd78e28731d87d56cc8ffef4a8f36ca26c38fe2de700543e627f8a464a"
 [[package]]
 name = "aziot-cert-client-async"
 version = "0.1.0"
-source = "git+https://github.com/Azure/iot-identity-service?branch=main#f6294bdd027284a27a03d83b4c5dc736d98e0a32"
+source = "git+https://github.com/Azure/iot-identity-service?branch=main#b59977078be65a4cc5d981cc036ced13f69b5f18"
```
Comment (vadim-kovalyov) [inline edgelet/Cargo.lock:90]: @arsing, let me know if I missed anything here.

Comment (arsing) [reply edgelet/Cargo.lock:90]: Yes, it's correct.

**Review by arsing — APPROVED** [review]

**Pull request merged**